Spigot-Server-Patches/0144-Reimplement-PlayerEditBookEvent.patch

```diff
@@ -0,0 +1,115 @@
+                     itemstack1 = this.player.getItemInMainHand();
+                     if (itemstack1 != null) {
+                         if (itemstack.getItem() == Items.WRITABLE_BOOK && itemstack1.getItem() == Items.WRITABLE_BOOK) {
++                            itemstack1 = new ItemStack(Items.WRITABLE_BOOK); // Paper
+                             itemstack1.a("author", (NBTBase) (new NBTTagString(this.player.getName())));
+                             itemstack1.a("title", (NBTBase) (new NBTTagString(itemstack.getTag().getString("title"))));
+                             NBTTagList nbttaglist = itemstack.getTag().getList("pages", 8);
+@@ -2270,12 +2275,14 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
+ 
+                             itemstack1.a("pages", (NBTBase) nbttaglist);
+                             itemstack1.setItem(Items.WRITTEN_BOOK);
++                            CraftEventFactory.handleEditBookEvent(player, itemstack1); // Paper
+                         }
+ 
+                         return;
+                     }
+                 } catch (Exception exception1) {
+                     PlayerConnection.LOGGER.error("Couldn\'t sign book", exception1);
++                    if (com.destroystokyo.paper.PaperConfig.invalidDataKick) this.disconnect("Invalid Book Data!"); // paper
+                     return;
+                 } finally {
+                     packetdataserializer.release();
+@@ -2341,6 +2348,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
+                         }
+                     } catch (Exception exception3) {
+                         PlayerConnection.LOGGER.error("Couldn\'t set command block", exception3);
++                        if (com.destroystokyo.paper.PaperConfig.invalidDataKick) this.disconnect("Invalid Command Block Data!"); // paper
+                     } finally {
+                         packetdataserializer.release();
+                     }
+@@ -2410,6 +2418,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
+                         }
+                     } catch (Exception exception4) {
+                         PlayerConnection.LOGGER.error("Couldn\'t set command block", exception4);
++                        if (com.destroystokyo.paper.PaperConfig.invalidDataKick) this.disconnect("Invalid Command Block Data!"); // paper
+                     } finally {
+                         packetdataserializer.release();
+                     }
+@@ -2435,6 +2444,7 @@ public class PlayerConnection implements PacketListenerPlayIn, ITickable {
+                                 }
+                             } catch (Exception exception5) {
+                                 PlayerConnection.LOGGER.error("Couldn\'t set beacon", exception5);
++                                if (com.destroystokyo.paper.PaperConfig.invalidDataKick) this.disconnect("Invalid Beacon Data!"); // paper
```

Comment:
If an error that never happens under normal conditions triggers, sounds reasonable to do...